File: source_code/python/solvewater_bfsgraph.py
from water import water
from bfsgraph import bfsgraph
from node import node

s0 = water(0, 0)
n, v = bfsgraph(s0)

if n:
    print("Solution")
    print("========")
    n.printsolution()
    print("========")
    print("Depth = %d, Cost = %d" % (n.depth, n.cost))
    print("No. of Visited Nodes = %d" % v)

File: source_code/java/SolveWaterBFSGraph.java
public class SolveWaterBFSGraph {
  public static void main(String[] args) {
    State s0 = new Water(0, 0);
    BFSGraph s = new BFSGraph();
    Node n = s.search(s0);
    if (n != null) {
      System.out.println("Solution");
      System.out.println("========");
      n.printSolution();
      System.out.println("========");
      System.out.println("Depth = " + n.depth + ", Cost = " + n.cost);
      System.out.println("No. of Visited Nodes = " + s.nvisited);
    }
  }
}
